tests/lib/eslint/flat-eslint.js

```diff
@@ -3813,6 +3813,26 @@ describe("FlatESLint", () => {
             assert.deepStrictEqual(rulesMeta, {});
         });
 
+        it("should not throw an error if results contain linted files and one ignored file", async () => {
+            const engine = new FlatESLint({
+                overrideConfigFile: true,
+                cwd: getFixturePath(),
+                ignorePatterns: "passing*",
+                overrideConfig: {
+                    rules: {
+                        "no-undef": 2,
+                        semi: 1
+                    }
+                }
+            });
+
+            const results = await engine.lintFiles(["missing-semicolon.js", "passing.js", "undef.js"]);
+            const rulesMeta = engine.getRulesMetaForResults(results);
```

Comment:
Okay, I've added an assertion to make sure that one of the messages returned by `lintFiles` belongs to an ignored file.
I chose this implementation because it doesn't require calling another method on the engine under test.
Alternatively, we could pass the `filePath` of each result to `isPathIgnored()`.

Reply:
This works. 👍